Sources/Common/Background Queue/Queue.swift

```diff
@@ -125,6 +141,31 @@ public class CioQueue: Queue {
         self.dateUtil = dateUtil
     }
 
+    // TODO: Write test case
+    public func getAllStoredTasks() -> [QueueTaskMetadata] {
+        storage.getInventory()
+    }
+
+    // TODO: Write test case
+    public func getTaskDetail(_ task: QueueTaskMetadata) -> (data: Data, taskType: QueueTaskType)? {
```

Comment:
we can probably get the timestamp from here?

```
        let createdAt = task.createdAt
```

Reply:
[Response](https://github.com/customerio/customerio-ios/pull/424#discussion_r1437854876)